template/.github/workflows/weekly_windows_macos.yml

```diff
@@ -0,0 +1,29 @@
+name: Windows and MacOS weekly tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * 1'
+
+jobs:
+  tests:
+    name: Tests
+    strategy:
+      matrix:
+        os: ['macos-latest', 'windows-latest']
+        python:
+          - version: '${{needs.formatting.outputs.min_python}}'
+            tox-env: '${{needs.formatting.outputs.min_tox_env}}'
```

Comment:
This won't work because there is no `needs`. We probably need an additional job or step to get the python version like we do in the `ci.yml` [workflow](https://github.com/scipp/copier_template/blob/main/template/.github/workflows/ci.yml#L19-L23):
```yaml
      - name: Get Python version for other CI jobs
        id: vars
        run: |
          echo "min_python=$(cat .github/workflows/python-version-ci)" >> $GITHUB_OUTPUT
          echo "min_tox_env=py$(cat .github/workflows/python-version-ci | sed 's/\.//g')" >> $GITHUB_OUTPUT
```